src/Catty/Functions/Text/RegularFunction.swift

```diff
@@ -0,0 +1,245 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+class RegularFunction: DoubleParameterStringFunction {
+    static var tag = "REGEX"
+    static var name = "regex"
+    static var defaultValue = "panda"
+    static var requiredResource = ResourceType.noResources
+    static var isIdempotent = true
+    static let position = 85
+
+    func tag() -> String {
+        return type(of: self).tag
+    }
+
+    func firstParameter() -> FunctionParameter {
+        return .string(defaultValue: " an? ([^ .]+)")
+    }
+
+    func secondParameter() -> FunctionParameter {
+        return .string(defaultValue: "I am a panda")
+    }
+
+    func value(firstParameter: AnyObject?, secondParameter: AnyObject?) -> String {
+        let pattern = type(of: self).interpretParameter(parameter: firstParameter)
+        let longText = type(of: self).interpretParameter(parameter: secondParameter)
+
+        let result = self.regularExpression(pattern: pattern, longText: longText)
```

Comment:
Looks like you spent a lot of time implementing this function! Thank you! However, I think you can simply use Apple's built-in RegEx functionality (e.g. https://developer.apple.com/documentation/foundation/nsregularexpression/1411510-rangeoffirstmatch) which might be more readable and faster.

Reply:
@m-herold  I have changed the regularExperission function, the logic is same as `rangeOfFirstMatch(in:options:range:)` but the approach is different. please review.